src/modules/dashboard/sagas/colony.ts

```diff
@@ -211,29 +163,17 @@ function* colonyAvatarUpload({
   payload: { colonyAddress, data },
 }: Action<ActionTypes.COLONY_AVATAR_UPLOAD>) {
   try {
-    // first attempt upload to IPFS
+    const apolloClient: ApolloClient<any> = yield getContext(
+      Context.APOLLO_CLIENT,
+    );
     const ipfsHash = yield call(ipfsUpload, data);
 
-    /*
-     * Set the avatar's hash in the store
-     */
-    yield executeCommand(setColonyAvatar, {
-      args: {
-        ipfsHash,
-      },
-      metadata: { colonyAddress },
-    });
-
-    /*
-     * Store the new avatar hash value in the redux store so we can show it
-     */
-    yield put<AllActions>({
-      type: ActionTypes.COLONY_AVATAR_UPLOAD_SUCCESS,
-      meta,
-      payload: { hash: ipfsHash },
+    yield apolloClient.mutate({
+      mutation: EditColonyProfileDocument,
+      variables: { input: { colonyAddress, avatarHash: ipfsHash } },
     });
   } catch (error) {
-    return yield putError(ActionTypes.COLONY_AVATAR_UPLOAD_ERROR, error, meta);
+    return yield putError(error.message, error, meta);
```

Comment:
We still need the UPLOAD_SUCCESS and UPLOAD_ERROR actions here for the async function creator to work